core/lib/server/pub-op.db.test.ts

```diff
@@ -0,0 +1,1331 @@
+			.execute();
+
+		const mainPub = await PubOp.create({
+			communityId: seededCommunity.community.id,
+			pubTypeId: seededCommunity.pubTypes["Basic Pub"].id,
+			lastModifiedBy: createLastModifiedBy("system"),
+		})
+			.set(seededCommunity.pubFields["Title"].slug, "Main pub")
+			.relate(seededCommunity.pubFields["Some relation"].slug, "relation 1", related1.id, {
+				replaceExisting: true,
+			})
+			.execute();
+
+		const updatedMainPub = await PubOp.update(mainPub.id, {
+			communityId: seededCommunity.community.id,
+			lastModifiedBy: createLastModifiedBy("system"),
+		})
+			.relate(seededCommunity.pubFields["Some relation"].slug, "relation 2", related2.id, {
+				replaceExisting: true,
+			})
+			.relate(
+				seededCommunity.pubFields["Some relation"].slug,
+				"relation 3",
+				PubOp.create({
+					communityId: seededCommunity.community.id,
+					pubTypeId: seededCommunity.pubTypes["Basic Pub"].id,
+					lastModifiedBy: createLastModifiedBy("system"),
+				}),
+				{ replaceExisting: true }
+			)
+			.execute();
+
+		// Should have relation 2 and 3, but not 1
+		expect(updatedMainPub).toHaveValues([
+			{ fieldSlug: seededCommunity.pubFields["Title"].slug, value: "Main pub" },
+			{
+				fieldSlug: seededCommunity.pubFields["Some relation"].slug,
+				value: "relation 2",
+				relatedPubId: related2.id,
+			},
+			{
+				fieldSlug: seededCommunity.pubFields["Some relation"].slug,
+				value: "relation 3",
+				relatedPubId: expect.any(String),
+			},
+		]);
+	});
+
+	it("should handle complex nested relation scenarios", async () => {
+		const trx = getTrx();
+		// manual rollback try/catch bc we are manually setting pubIds, so a failure in the middle of this will leave the db in a weird state
+		try {
+			// Create all pubs with meaningful IDs
+			const pubA = "aaaaaaaa-0000-0000-0000-000000000000" as PubsId;
+			const pubB = "bbbbbbbb-0000-0000-0000-000000000000" as PubsId;
+			const pubC = "cccccccc-0000-0000-0000-000000000000" as PubsId;
+			const pubD = "dddddddd-0000-0000-0000-000000000000" as PubsId;
+			const pubE = "eeeeeeee-0000-0000-0000-000000000000" as PubsId;
+			const pubF = "ffffffff-0000-0000-0000-000000000000" as PubsId;
+			const pubG = "11111111-0000-0000-0000-000000000000" as PubsId;
+			const pubH = "22222222-0000-0000-0000-000000000000" as PubsId;
+			const pubI = "33333333-0000-0000-0000-000000000000" as PubsId;
+			const pubJ = "44444444-0000-0000-0000-000000000000" as PubsId;
+			const pubK = "55555555-0000-0000-0000-000000000000" as PubsId;
+			const pubL = "66666666-0000-0000-0000-000000000000" as PubsId;
+
+			// create the graph structure:
+			//          A          J
+			//       /     \       |
+			//     /        \      |
+			//   B           C --> I
+			//   |         /  \
+			//   G -->  E      D
+			//               /  \
+			//             F     H
+			//                 /  \
+			//                K --> L
+
+			// create leaf nodes first
```

Comment:
this is by far the most complex test, and very curious whether you agree with whether this should be the behavior.
i spent a bit too long trying to figure this out haha

see my comment in `pub-op.ts` for a more in depth explanation about the "algorithm" for determining orphans